benchmarks/cluster/ArborX_DBSCANVerification.hpp

```diff
@@ -261,80 +279,205 @@ bool verifyClustersAreUnique(ExecutionSpace const &exec_space,
       static_cast<Verify>(verifyBorderAndNoisePoints),
       static_cast<Verify>(verifyClustersAreUnique)};
   return std::all_of(verify.begin(), verify.end(), [&](Verify const &verify) {
-    return verify(exec_space, indices, offset, labels, core_min_size, verbose);
+    return verify(exec_space, offset, neighbors, verbose);
   });
 }
 
-struct IndexOnlyCallback
+#ifdef ARBORX_ENABLE_MPI
+struct IndexCallback
+{
+  template <typename Query, typename Value, typename Output>
+  KOKKOS_FUNCTION auto operator()(Query const &, Value const &value,
+                                  Output const &out) const
+  {
+    out({value.index});
+  }
+};
+
+template <typename Labels, typename IsCore>
+struct IndexRankLabelCoreCallback
 {
+  int _rank;
+  Labels _labels;
+  IsCore _is_core;
+
   template <typename Query, typename Value, typename Output>
   KOKKOS_FUNCTION auto operator()(Query const &, Value const &value,
                                   Output const &out) const
   {
-    out(value.index);
+    auto const i = value.index;
+    out({i, _rank, _labels(i), _is_core(i)});
   }
 };
 
-template <typename ExecutionSpace, typename Primitives, typename LabelsView,
+template <typename ExecutionSpace, typename Primitives, typename Labels,
           typename Coordinate>
-bool verifyDBSCAN(ExecutionSpace exec_space, Primitives const &primitives,
-                  Coordinate eps, int core_min_size, LabelsView const &labels,
-                  bool verbose = false)
+bool verifyDBSCAN(MPI_Comm comm, ExecutionSpace exec_space,
+                  Primitives const &primitives, Coordinate eps,
+                  int core_min_size, Labels const &labels, bool verbose = false)
 {
-  Kokkos::Profiling::pushRegion("ArborX::DBSCAN::verify");
+  Kokkos::Profiling::ScopedRegion guard("ArborX::DBSCAN::verify");
 
-  static_assert(Kokkos::is_view<LabelsView>{});
+  static_assert(Kokkos::is_view<Labels>{});
 
   using Points = Details::AccessValues<Primitives>;
   using MemorySpace = typename Points::memory_space;
 
-  static_assert(std::is_same<typename LabelsView::value_type, int>{});
-  static_assert(std::is_same<typename LabelsView::memory_space, MemorySpace>{});
+  static_assert(std::is_same_v<typename Labels::value_type, int>);
+  static_assert(std::is_same_v<typename Labels::memory_space, MemorySpace>);
 
   ARBORX_ASSERT(eps > 0);
   ARBORX_ASSERT(core_min_size >= 2);
 
   Points points{primitives}; // NOLINT
+  auto const n = points.size();
 
   using Point = typename Points::value_type;
   static_assert(GeometryTraits::is_point_v<Point>);
 
-  ArborX::BoundingVolumeHierarchy bvh(
-      exec_space, ArborX::Experimental::attach_indices(points));
+  // FIXME_NVCC: nvcc 11.7 (and maybe others) cannot compile this without
+  // template arguments.
+  ArborX::DistributedTree<MemorySpace,
+                          PairValueIndex<typename Points::value_type, unsigned>>
+      index(comm, exec_space, ArborX::Experimental::attach_indices(points));
```

Comment:
The `DistributedTree`'s deduction guide cannot be compiled here by NVCC. In other places it compiles. No idea why, but not spending time to investigate.